src/memory.js

```diff
@@ -1,18 +1,35 @@
 class MemoryGame {
   constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
   }
 
   shuffleCards() {
-    // ... write your code here
+    if (!this.cards) {
+      return undefined;
+    } else {
+      return this.cards.sort(() => Math.random() - 0.5);
+    }
   }
 
   checkIfPair(card1, card2) {
-    // ... write your code here
+    this.pairsClicked++;
+
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true
+    } else {
+      return false;
+    }
   }
 
   checkIfFinished() {
-    // ... write your code here
+    if (this.pairsGuessed === this.cards.length / 2) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
Esto si devuelves directamente la condicion del if funciona igual! :D

```suggestion
    return this.pairsGuessed === this.cards.length / 2;
```